src/pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -525,3 +533,19 @@ def _detect_light_colab_theme() -> bool:
         except ModuleNotFoundError:
             return False
     return False
+
+
+def _is_jupyter() -> bool:
+    """Detect if it's in a Jupyter environment."""
+    try:
+        get_ipython  # type: ignore
+    except NameError:
+        return False
+
+    shell_name = get_ipython().__class__.__name__  # noqa: F821
+
+    # https://stackoverflow.com/questions/15411967/how-can-i-check-if-code-is-executed-in-the-ipython-notebook/
+    if shell_name == "ZMQInteractiveShell":
```

Comment:
I confirmed this condition works in a Jupyter notebook (where it should be `True`) and in Google Colab (where it should be `False`), but I am not too confident because I couldn't find any official documentation on this.

Reply:
@kaushikb11 @carmocca What do you think?

Reply:
LGTM. Why does colab not need this?